astropy/time/core.py

```diff
@@ -2275,6 +2351,44 @@ def __add__(self, other):
     def __radd__(self, other):
         return self.__add__(other)
 
+    def mean(self, axis=None, dtype=None, out=None, keepdims=False, *, where=True):
+
+        scale = self.scale
+        if scale == 'utc':
+            self = self.tai
+        result = super().mean(axis=axis, dtype=dtype, out=out, keepdims=keepdims, where=where)
+        if scale == 'utc':
+            result = result.utc
+
+        result.out_subfmt = self.out_subfmt
+
+        location = self.location
+        if self.location is not None:
+            if self.location.shape:
+
+                if axis is None:
+                    axis_normalized = tuple(range(self.ndim))
+                elif isinstance(axis, int):
+                    axis_normalized = axis,
+                else:
+                    axis_normalized = axis
+
+                sl = [slice(None)] * self.location.ndim
+                for a in axis_normalized:
+                    sl[a] = slice(0, 1)
+
+                if np.any(self.location != self.location[tuple(sl)]):
+                    raise ValueError("`location` must be constant over the reduction axes.")
+
+                if not keepdims:
+                    for a in axis_normalized:
+                        sl[a] = 0
```

Comment:
Maybe good to have a test that checks that `location` is dealt with correctly?

Reply:
I sort of test this on lines 670-673 of `astropy/time/tests/test_methods.py`. Do you think that is sufficient, or should I add more tests?

Reply:
Coverage states that it is not covered - that's why I suggested to test both `keepdims` possibilities below.